cluster/node.go

```diff
@@ -0,0 +1,103 @@
+package cluster
+
+import (
+	"encoding/json"
+	"sync"
+	"time"
+)
+
+//go:generate stringer -type=NodeState
+type NodeState int
+
+const (
+	NodeReady NodeState = iota
+	NodeNotReady
+)
+
+type Node struct {
+	sync.RWMutex
+	name          string
+	version       string
+	primary       bool
+	primaryChange time.Time
+	state         NodeState
+	started       time.Time
+	stateChange   time.Time
```

Comment:
I think we should make clear which fields are protected by the mutex and which are not, like so:

```
type Node struct {
	name          string
	version       string
	started       time.Time

	sync.RWMutex
	primary       bool
	primaryChange time.Time
	state         NodeState
	stateChange   time.Time
}
```